spring-pulsar/src/main/java/org/springframework/pulsar/config/GenericPulsarListenerContainerFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.config;
+
+/**
+ * Factory for Pulsar message listener containers.
+ *
+ * @param <C> message listener container type.
+ * @param <E> listener endpoint type.
+ * @author Soby Chacko
+ * @author Christophe Bornet
+ */
+public interface GenericPulsarListenerContainerFactory<C, E extends GenericPulsarListenerEndpoint> {
+
+	C createListenerContainer(E endpoint);
```

Comment:
I think javadoc would help on these 2 methods to understand the naming difference (one is a "listener" container - but they are both type `C`).
It is a bit confusing as-is.

Reply:
I think this can be simplified if we introduce a `MessageListenerContainer extends Disposable, SmartlifeCycle` like in spring-kafka. Will give it a shot.

Reply:
Done.